src/internal.c

```diff
@@ -12007,15 +12032,13 @@ int ProcessPeerCerts(WOLFSSL* ssl, byte* input, word32* inOutIdx,
             goto exit_ppc;
     }
     else
-#elif defined(WOLFSSL_NONBLOCK_OCSP)
-    if (args == NULL) {
-        args = (ProcPeerCertArgs*)XMALLOC(
-            sizeof(ProcPeerCertArgs), ssl->heap, DYNAMIC_TYPE_TMP_BUFFER);
-        if (args == NULL) {
-            ERROR_OUT(MEMORY_E, exit_ppc);
-        }
+#endif
+#ifdef WOLFSSL_NONBLOCK_OCSP
+    if (ssl->error == OCSP_WANT_READ) {
+        /* Re-entry after non-blocking OCSP */
     }
-    if (ssl->nonblockarg == NULL) /* new args */
+    else
+#endif
```

Comment:
[can't put the comment on the right lines because they aren't part of the diff. lines 12045
Is the 
```
if (args == NULL) {
        ERROR_OUT(MEMORY_E, exit_ppc);
}
```
check outside of the else branch of line 12040 on purpose?
I don't think it hurts, but the indentation is misleading.

Reply:
The `else` applies to the `/* Reset state */` block. The block you cited is in the `elif` when building for small stack.